frontend/insurance/src/app/core/services/auth/auth-guard.service.ts

```diff
@@ -1,18 +1,25 @@
 import { Injectable } from '@angular/core';
 import { AuthService } from './auth.service';
-import { Router } from '@angular/router';
+import { Router, ActivatedRouteSnapshot } from '@angular/router';
 
 @Injectable({
   providedIn: 'root',
 })
 export class AuthGuardService {
   constructor(public authService: AuthService, public router: Router) {}
 
-  canActivate(): boolean {
+  canActivate(
+    route: ActivatedRouteSnapshot
+  ): boolean {
     if (!this.authService.isLoggedIn) {
       if (window.localStorage.getItem('auth')) {
         return true;
       }
+
+      localStorage.setItem('url', route.url.join('/'));
```

Comment:
Why?

The only reason why the URL isn't persisting is because you have the AuthGuard service listed for some routes. Just remove the AuthGuard for the route and instead use it for showing/hiding buttons.

Reply:
When logging in it always redirects you to the homepage. If you have only conditional logic on edit buttons, and then you login on the edit claim page, it will redirect you to homepage if we do not store the requested URL (on refresh it would be the current URL).